File: Part 2 - Regression/Section 8 - Decision Tree Regression/arbolesDecision.R
#Arbol de decision para regresion
#install.packages("rpart")
library(rpart)
library(ggplot2)

dataset = read.csv("Position_Salaries.csv")

dataset = dataset[,2:3]


reg_tree = rpart(formula = Salary ~ .,
                 data = dataset,
                 control = rpart.control(minsplit = 1))


y_pred = predict(reg_tree, newdata= data.frame(Level = 6.5))


x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() + 
  geom_point(aes(x = dataset$Level , y = dataset$Salary),
             color = "red") +
  geom_line(aes(x= x_grid), y = predict(reg_tree, newdata = data.frame(Level = x_grid)),
            color = "blue") +
  ggtitle("Prediccion arbol de decision del sueldo en funcion del nivel del empleado") +
  xlab("Nivel del empleado") +
  ylab("Sueldo (en $)")

File: Part 2 - Regression/Section 8 - Decision Tree Regression/ArbolesDecision.py
# -*- coding: utf-8 -*-
"""
Created on Sat Nov 23 19:56:31 2019

@author: darub
"""
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeRegressor
dataset = pd.read_csv("Position_Salaries.csv")

x = dataset.iloc[:,1:-1].values
y = dataset.iloc[:,-1].values


# Entrenar el modelo, random_state generar semilla con valor 0
reg_tree = DecisionTreeRegressor(random_state= 0)
reg_tree.fit(x,y)

# Prediccion
y_pred = reg_tree.predict([[6.5]])


# Graficar modelo, divide los datos en segmentos si es mayor que 10 siempre dara el valor del salario en 10
x_grid = np.arange(min(x), max(x), 0.1) 
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x,y, color="red")
plt.plot(x_grid , reg_tree.predict(x_grid), color="blue")
plt.title("Modelo arboles de decision por regresion")
plt.xlabel("Posicion del empleado años")
plt.ylabel("Sueldo (en $)")
plt.show()














